packages/material-ui/src/Modal/Modal.test.js

```diff
@@ -375,6 +375,40 @@ describe('<Modal />', () => {
       const modalNode = modalRef.current;
       assert.strictEqual(modalNode.getAttribute('aria-hidden'), 'true');
     });
+
+    /* Test case for https://github.com/mui-org/material-ui/issues/15180 */
+    it('should remove the transition children in the DOM when closed on whilst transition status is entering', () => {
+      const TestCaseChildren = props => (
+        <Fade in={props.open} tabIndex={-1}>
+          <span>Hello</span>
+        </Fade>
+      );
+      TestCaseChildren.propTypes = { open: PropTypes.bool };
+
+      const TestCase = props => (
+        <Modal open={props.open} keepMounted={false}>
+          <TestCaseChildren open={props.open} />
+        </Modal>
+      );
+      TestCase.propTypes = { open: PropTypes.bool };
+
+      const wrapper = mount(<TestCase open={false} />);
+      assert.strictEqual(wrapper.contains(TestCaseChildren), false);
+
+      wrapper.setProps({ open: true });
+      wrapper.update();
+      assert.strictEqual(wrapper.contains(TestCaseChildren), true);
+      assert.strictEqual(
+        wrapper
+          .find(TestCaseChildren)
+          .find('Transition')
+          .instance().state.status,
```

Comment:
We are moving away from testing internals.

Reply:
👍 Removing this will still test and assert what is really required.
The same should apply to the Popper test (https://github.com/mui-org/material-ui/blob/fdbc757d6dd5eaf8da5a5f5f96062257ec77deaa/packages/material-ui/src/Popper/Popper.test.js#L190), I presume?

Reply:
What I meant is no `.instance()` usage.

Reply:
The child of the Transition component has access to current state/phase of the transition via render props. You could forward this to the DOM via text or attribute and assert on that.